bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -1509,7 +1513,10 @@ void ReplicaImp::onMessage<ReplicaStatusMsg>(ReplicaStatusMsg *msg) {
 
       for (SeqNum i = beginRange; i <= endRange; i = i + checkpointWindowSize) {
         CheckpointMsg *checkMsg = checkpointsLog->get(i).selfCheckpointMsg();
-        if (checkMsg != nullptr) send(checkMsg, msgSenderId);
+        if (checkMsg != nullptr) {
+          send(checkMsg, msgSenderId);
+          metric_sent_checkpoint_msg_due_to_status_.Get().Inc();
```

Comment:
You use the same pair of operators again and again: send() and increase of some metric.
I propose wrapping send() function for this purpose and pass the corresponding metric (CounterHandle) to it.
This will safe a lot of additional code lines.

Reply:
I would prefer to leave it as it is.
I want to see explicitly where the metrics are taken and taking metrics is not part of "send" functionality...
wrapping it in send may be confusing